### PR TITLE
feat: House Consumption Today + correct the load mislabel (modbus 2.1.1)

### PR DESCRIPTION
**Pull request by dewet22**
## Summary

Adopts **givenergy-modbus 2.1.1** (#174), which corrected two single-phase field mislabels and added the real derived house-consumption figure. This fixes the dashboard's long-standing consumption cliff — givenergy_local showed ~0 consumption because the old "Load Energy Today" sensor read IR35, which is actually AC charge, not house load.

## Changes

- **New "House Consumption Today" sensor** (`e_consumption_today`) — the GE-app "Consumption today" derived value (PV generation + grid-in − grid-out − AC-charge), single-phase only (defensive getattr + `skip_if_none`). The dashboard "Consumed" series now points here; `DASHBOARD_VERSION` 5 → 6 so existing users are prompted to regenerate.
- **Renamed the mislabelled load sensor**: `e_load_day` → `e_ac_charge_today` ("AC Charge Today"). A **unique_id migration** (`_migrate_unique_ids` in `__init__.py`) re-points the existing entity in place so its history/statistics survive rather than orphaning it.
- **Held the inverter-output pair**: `e_inverter_out_day` entity (key/name) unchanged — the rename is deferred until the matching *total* (IR45/46) is verified and renamed library-side — but its `value_fn` reads the renamed `e_pv_generation_today` directly to avoid a per-poll DeprecationWarning.
- Pin → `givenergy-modbus>=2.1.1`.
- **GivTCP migration script**: resilient recorder writes (chunk large imports, retry on HA's command-timeout, pace entities — fixes the timeout cascade that left entities cleared-but-not-reimported mid-run), and the house-consumption pair retargeted at the real `house_consumption_today` sensor.

## Notes

- The `ha-entity-reviewer` agent confirmed the unique_id migration is correct on every edge case (slicing, multi-inverter serials, collision handling, ordering before platform setup).
- 188 tests pass (incl. a migration round-trip test); mypy at the established baseline.
- Inverter-output entity rename is **held** pending a modbus handoff to confirm/rename `e_inverter_out_total` (IR45/46) so today+total move together.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added "House Consumption Today" and "AC Charge Today" sensors
  * Updated generated dashboards to display consumption from the new sensor

* **Bug Fixes**
  * Automatic entity unique-id migration preserves entity history and avoids collisions during sensor renames

* **Documentation**
  * Updated migration guide to reflect new sensor mappings

* **Tests**
  * Added tests covering new sensors and migration behavior

[Review Change Stack](https://app.coderabbit.ai/change-stack/dewet22/givenergy-hass/pull/111?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by dewet22-codex** [comment]
I’ve started reviewing this PR.

File: scripts/migrate_from_givtcp.py [hunk 1]
```diff
@@ -52,17 +52,19 @@
...
   Inverter output today / lifetime
-  House load today
   Battery throughput lifetime
   Battery charge cycles (per battery pack)
 
 Opt-in (--include-charge-from-grid):
   Charge from grid lifetime  ⚠️  values differ on some systems — verify manually
 
-Not migrated (no GivTCP equivalent or register-level gap):
+Not migrated:
```
Comment (dewet22-codex) [inline scripts/migrate_from_givtcp.py]: I think this help text needs updating alongside the new migration pair below. `INVERTER_PAIRS` now includes `load_energy_today_kwh -> house_consumption_today` by default, so running the script with `--help` or without opening the source still says house load is not migrated even though it now is. I’d move the new house-consumption wording into the default migrated list and leave only genuinely unsupported sensors here.

File: scripts/migrate_from_givtcp.py [hunk 2]
```diff
@@ -393,11 +471,13 @@ async def migrate_entity(
 # Main
 # ---------------------------------------------------------------------------
 
+
 async def run(args: argparse.Namespace) -> int:
```
[the remark this answers is not in the document]
Comment (dewet22) [reply scripts/migrate_from_givtcp.py:561]: Addressed in 446231f — extracted the cutover-suggestion print, plan construction, and summary table into helpers; `run()` drops from ~161 to ~91 lines. The battle-tested execute/write path (`migrate_entity` + the apply loop) is untouched.

File: scripts/migrate_from_givtcp.py [hunk 3]
```diff
@@ -74,7 +76,7 @@
 import json
 import re
 import sys
-from datetime import date, datetime, timedelta, timezone
+from datetime import UTC, date, datetime, timedelta
 from typing import Any
```
[the remark this answers is not in the document]
Comment (dewet22) [reply scripts/migrate_from_givtcp.py:81]: Declining — `asyncio` *is* imported (line 75). The script entrypoint is `asyncio.run(run(args))` and the live migration runs exercised `asyncio.sleep` without a NameError, so this looks like a parser false positive rather than a real missing import.

File: custom_components/givenergy_local/sensor.py [hunk 4]
```diff
@@ -490,20 +490,43 @@ class GivEnergyCoordinatorSensorDescription(SensorEntityDescription):
         value_fn=lambda inv: inv.p_load_demand,
     ),
     GivEnergyInverterSensorDescription(
-        key="e_load_day",
-        name="Load Energy Today",
+        # Derived house consumption (single-phase only) — matches the GE app's
+        # "Consumption today". Single-phase inverters expose no consumption
+        # register; givenergy-modbus computes it (PV gen + grid-in − grid-out −
+        # AC-charge). Three-phase has no such field, so skip_if_none drops it
+        # there (and the value_fn getattr keeps it None-safe).
```
[the remark this answers is not in the document]
Comment (dewet22) [reply custom_components/givenergy_local/sensor.py:497]: Fixed in 446231f — replaced the U+2212 minus signs with ASCII hyphens in the consumption-formula comments (sensor.py and the migrate script).

File: scripts/migrate_from_givtcp.py (hunk 1 above)
Comment (dewet22) [reply scripts/migrate_from_givtcp.py]: Fixed in 723710c — moved "House consumption today" into the default-migrated list and dropped the stale "Not migrated: House load today" entry, so `--help` now matches `INVERTER_PAIRS`.

File: scripts/migrate_from_givtcp.py [hunk 5]
```diff
@@ -393,11 +471,99 @@ async def migrate_entity(
...
+    for sn in batt_serials:
+        for gt_sfx, ge_sfx, desc, fallback_unit in BATTERY_PAIRS:
+            givtcp_id = f"sensor.givtcp_{sn}_{gt_sfx}"
+            ge_id = f"sensor.givenergy_battery_{sn}_{ge_sfx}"
+            unit = meta_by_id.get(ge_id, {}).get("unit_of_measurement") or fallback_unit
+            plan.append((givtcp_id, ge_id, desc, unit, False))
+    return plan
+
+
+def _print_summary(results: list[MigrationResult], applying: bool) -> int:
```
[the remark this answers is not in the document]
Comment (dewet22) [reply scripts/migrate_from_givtcp.py:527]: Addressed in 723710c — replaced `_print_summary`'s four sum() comprehensions with a single `Counter`, dropping its cyclomatic complexity below the gate.

**Comment by dewet22-codex** [comment]
I’ve started re-reviewing this PR.

**Review by dewet22-codex — APPROVED** [review]
Approved. I re-reviewed the current head and do not have any actionable concerns. I checked the updated diff and current GitHub checks; the earlier migration-script help-text finding has been addressed.
